src/components/proposal/ProposalContainer.tsx

```diff
@@ -150,17 +151,37 @@ export default function ProposalContainer(props: {
     }
   }, [previousInitialProposal, props.proposal, dispatch]);
 
+  const hasReferenceNumberFormat = state.proposal.call?.referenceNumberFormat
+    ? true
+    : false;
+
+  const isSubmitted = state.proposal.submitted;
+
+  const proposalId = state.proposal.proposalId;
+
+  let info: JSX.Element | string = 'DRAFT';
```

Comment:
My suggestion is also to simplify here a bit to something like: `let info: JSX.Element | string = proposalId || 'DRAFT';` and after that continue with: 
`
if (!isSubmitted && hasReferenceNumberFormat) {
      info = (
        <Typography>
          {proposalId}
          <br />
          <small>Pre-submission reference</small>
        </Typography>
      );
    }
`
I think it will do the same but without nesting if blocks. What you have here is also correct but it's just my suggestion to make it more readable.